src/lug/Main/MainAndroid.cpp

```diff
@@ -0,0 +1,126 @@
+#include <lug/Main/MainAndroid.hpp>
+
+namespace lug {
+namespace Main {
+
+
+LugAndroidApp::LugAndroidApp(ANativeActivity* activity, void* savedState, size_t savedStateSize)
+    : _activity(activity), _savedState(nullptr), _savedStateSize(0) {
+    activity->callbacks->onInputQueueCreated = &LugAndroidApp::onInputQueueCreated;
+    activity->callbacks->onInputQueueDestroyed = &LugAndroidApp::onInputQueueDestroyed;
+
+    activity->callbacks->onNativeWindowCreated = &LugAndroidApp::onNativeWindowCreated;
+    activity->callbacks->onNativeWindowDestroyed = &LugAndroidApp::onNativeWindowDestroyed;
+    activity->callbacks->onNativeWindowResized = &LugAndroidApp::onNativeWindowResized;
+    activity->callbacks->onNativeWindowRedrawNeeded = &LugAndroidApp::onNativeWindowRedrawNeeded;
+    activity->callbacks->onContentRectChanged = &LugAndroidApp::onContentRectChanged;
+
+    activity->callbacks->onWindowFocusChanged = &LugAndroidApp::onWindowFocusChanged;
+    activity->callbacks->onConfigurationChanged = &LugAndroidApp::onConfigurationChanged;
+    activity->callbacks->onLowMemory = &LugAndroidApp::onLowMemory;
+    activity->callbacks->onStart = &LugAndroidApp::onStart;
+    activity->callbacks->onResume = &LugAndroidApp::onResume;
+    activity->callbacks->onSaveInstanceState = &LugAndroidApp::onSaveInstanceState;
+    activity->callbacks->onPause = &LugAndroidApp::onPause;
+    activity->callbacks->onStop = &LugAndroidApp::onStop;
+    activity->callbacks->onDestroy = &LugAndroidApp::onDestroy;
+
+    if (savedState != nullptr) {
+        _savedState = ::operator new(savedStateSize);
+        _savedStateSize = savedStateSize;
+        memcpy(_savedState, savedState, savedStateSize);
+    }
+
+    std::thread t(&LugAndroidApp::startApplication, this);
+    t.detach();
+}
+
+
+void LugAndroidApp::shutdowApplication() {
+    _mutex.lock();
+    if (_savedState != nullptr) {
+        ::operator delete(_savedState);
+        _savedState = nullptr;
+        _savedStateSize = 0;
+    }
+    _mutex.unlock();
+    delete this;
```

Comment:
How would you implement it then? We have to alloc it to java in ANativeActivity_onCreate and it is guaranteed that shutdowApplication is in fact, putting the application down for real. If we don't delete ourselves, we leak some memory